### PR TITLE
Update 'formattedDate' if Date Picker's 'value' is changed via props.

### PR DESCRIPTION
**Pull request by luhu**
This is the fix for https://github.com/OfficeDev/office-ui-fabric-react/issues/20

### CODE REVIEW

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -85,6 +85,16 @@ export class DatePicker extends React.Component<IDatePickerProps, IDatePickerSta
     this._validateTextInput = this._validateTextInput.bind(this);
   }
 
+  public componentWillReceiveProps(nextProps: IDatePickerProps) {
+    let { formatDate, value } = nextProps;
+
+    this.setState({
+      selectedDate: value || new Date(),
+      formattedDate: formatDate && value ? formatDate(value) : null,
+      errorMessage: ''
```
Comment (dzearing) [inline src/components/DatePicker/DatePicker.tsx]: do you really intend to clear the errorMessage whenever new props are passed in? That seems a little odd.

Comment (luhu) [inline src/components/DatePicker/DatePicker.tsx]: As a new valid value (a Date object or null/undefined ensured by Typescript) is passed via props, we shall clean errorMessage here.

Comment (dzearing) [inline src/components/DatePicker/DatePicker.tsx]: ...right, but wouldn't you want to *re-evaluate* the error state in this case? What if a blank date is passed in and isRequired is true?

Comment (luhu) [inline src/components/DatePicker/DatePicker.tsx]: You are right, I updated the PR with the errorMessage state re-evaluation logic. Thanks!

**Comment by dzearing** [comment]
Approved

**Pull request merged**